cmd/acr/purge.go

```diff
@@ -264,6 +270,21 @@ func getTagsToDelete(ctx context.Context,
 				// If a tag does not match the regex then it not added to the list no matter the LastUpdateTime
 				continue
 			}
+			isExcluded := false
+			for _, exFilter := range excludeFilter {
+				exFilterRegex, exFilterErr := regexp.Compile(exFilter)
```

Comment:
`exFilter` is complied unnecessarily multiple times. You should pre-compile it like `filter`.